### ISSUE TITLE
DOC,DISCUSS: Reorganizing documentation development and contribution docs

### ISSUE THREAD

**Issue by Mukulikaa** [numpy/numpy#19606]
We have multiple documents for documentation development- [How to contribute to the NumPy documentation](https://numpy.org/devdocs/dev/howto-docs.html), [NumPy’s Documentation](https://numpy.org/devdocs/docs/index.html), [Documentation conventions](https://numpy.org/devdocs/doc_conventions.html). This issue is to discuss ways to merge these documents if possible and figure out where they should live.

**Comment by rossbar** [comment]
**tl;dr - my opinions ordered by strength of opinion:**
 1. Remove the entire [documentation conventions page](https://numpy.org/devdocs/doc_conventions.html)
 2. Move the [User documentation section](https://numpy.org/devdocs/docs/howto_document.html#user-documentation) from the user guide to the [contributor guide](https://numpy.org/devdocs/dev/howto-docs.html)
 3. Remove the entire [Guide to NumPy Documentation page](https://numpy.org/devdocs/docs/howto_document.html#a-guide-to-numpy-documentation) page from the user guide; or move it to the contributor guide.

---

IMO the [Documentation conventions](https://numpy.org/devdocs/doc_conventions.html#documentation-conventions) page is redundant and I'd be inclined to remove it entirely. If there's pushback on removing it, I think it should at least be moved to the contributor guide.

I also think that the [User documentation](https://numpy.org/devdocs/docs/howto_document.html#user-documentation) section that is currently in the user guide would make a lot more sense if it were in the [documentation contributor guide](https://numpy.org/devdocs/dev/howto-docs.html), perhaps under a "Style" heading, as it only really deals with writing docs.

The same could arguable said for the rest of the [Guide to NumPy page](https://numpy.org/devdocs/docs/howto_document.html#a-guide-to-numpy-documentation). I wouldn't be against folding the entire page into the contributors docs or even removing it entirely in favor of a link to the `numpydoc` documentation, which is more focused and does a better job explaining the numpydoc docstring format.

**Comment by Mukulikaa** [comment]
I agree that we could remove the Documentation conventions page since most of it is already covered in other parts of the documentation. If required we could add the line 

> This is not part of the code and will cause an error if typed or pasted into the Python shell. 

to the [Reading the example code](https://numpy.org/devdocs/user/absolute_beginners.html#reading-the-example-code) section.

>  I wouldn't be against folding the entire page into the contributors docs

I like this idea too; I think it makes sense.

I was also wondering if we should move the [Building the NumPy API and reference docs](https://numpy.org/devdocs/docs/howto_build_docs.html) page to the Development section. I don't think users usually build the documentation locally?

**Comment by rossbar** [comment]
> I was also wondering if we should move the Building the NumPy API and reference docs page to the Development section. I don't think users usually build the documentation locally?

Good point, this makes sense to me as well, and is nice because it the results in all of these documents being removed from the user guide (where their fit is questionable). I'd be +1 for this change.
